theano/tensor/nnet/nnet.py

```diff
@@ -1585,9 +1488,11 @@ def local_advanced_indexing_crossentropy_onehot(node):
         # Check that rows == arange(labels.shape[0])
         if _check_rows_is_arange_len_labels(rows, labels):
             if labels.ndim == 1 and x_var.ndim == 2:
-                return [-crossentropy_softmax_argmax_1hot_with_bias(x_var,
-                                                                    b_var,
-                                                                    labels)[0]]
+                ret = -crossentropy_softmax_argmax_1hot_with_bias(x_var,
+                                                                  b_var,
+                                                                  labels)[0]
+                copy_stack_trace([x_var, b_var, labels], ret)
```

Comment:
This would actually copy the stack trace to the `neg` node that corresponds to the minus sign.
I'm not sure what would be best to assign to the output of `crossentropy_...bias`, but I think the stack of `node.outputs[0]` would make sense for both, even if they do not completely correspond to the computation being performed. Ideally, we would also have the intermediate steps of the computation as well (corresponding to `softmax(x)` and `log(softmax(x))` in the first case, and to `softmax(x)` and `softmax(x)[...]` in the second case), but it would require some bigger changes in the optimization itself.
So for the moment, I think `copy_stack_trace(node.outputs[0], [minus_ret, ret])` (where `minus_ret` would be `crossentropy_...(...)[0]`) would be enough.